[1.16] Adds workflow limitations (#4865)
dapr/docs: daprdocs/content/en/developing-applications/building-blocks/workflow/workflow-architecture.md

Workflow state can be purged from a state store, including all its history.
Each Dapr SDK exposes APIs for purging all metadata related to specific workflow instances.

#### State store record count

> The number of records which will be saved as history in the state store per workflow run is determined by its complexity or "shape", i.e. the number of activities, timers, sub-workflows etc.

Reviewer: ```suggestion
The number of records which are saved as history in the state store per workflow run is determined by its complexity or "shape". In other words, the number of activities, timers, sub-workflows etc.
```
